site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+
+### Status additions
+
+The basic design:
+- we'll add a `grandparents` stanza inside the parents stanza in status, which will contain details about the route being a parent, successfully or not.
+- the `grandparents` stanza will only be populated if a rule has a non-nil `allowedRoutes` section.
+- once the `grandparents` stanza is populated, Conditions are used to describe if the configuration still generates a valid Ephemeral Route configuration.
+
+Here's what the proposal looks like for an attached child route:
+
+```yaml
+status:
+  # Existing stanza is the same
+  parents:
+  - parentref:
+      name: somegateway
+      namespace: somenamespace
+    controllerName: controller.io/somecontroller
+    conditions:
+    - <conditions here>
+    grandparents:
+    - conditions:
+      - <conditions here>
+```
+We add a new `grandparents` stanza inside the the top-level `parents` status. This stanza will be `+optional` and `omitempty`, with the following rules:
+- `grandparents` will be nil in all non-parent HTTPRoutes.
+- Once a HTTPRoute is a parent, then this stanza must also be non-nil.
+
+We need to add this inside the parents stanza because it's possible that a parent HTTPRoute may be attached to more than one parent (probably Gateways) of its own, and there may be different status between different Gateways. For example, if a parent HTTPRoute is attached to two Gateways, one of which has a wildcard hostname, whether or not the child HTTPRoute attempting to attach to the parent HTTPRoute will be accepted is dependent on the hostname setting on the Gateway Listener. An example demonstrating this is included below.
+
+The struct itself has the same `parentRef` and `conditions` fields as the `parents` struct.
+
+The `parentRef` in the `grandparents` struct must be the same as the one on the *parent* Route's status. See the example for more clarity here.
+
+For Conditions:
+- when the controller first sees the parent Route, it must add the following Condition Types to the *parent* Route:
+  - `Attached`: one or more Routes have successfully attached to this parent. False if zero Routes have attached successfully. Note that a new Reason has also been added, `PartiallyAttached` which indicates that *some* routes have tried to attach and failed.
+  - `Conflicted`: one or more settings in the parent conflict with settings in one of the child routes. The message should contain more details. Routes that cause this to be true do not attach.
+  - `ResolvedRefs`: one or more Routes have failed to attach because there is a problem with their reference. (for example, they could not be found). Those routes do not attach.
+  - `NotSupported`: the controller that is reconciling this Route does not support attaching at least one of the routes that has attempted to attach, or the parent route is already a child route itself. Those routes do not attach.
+
+It must also add the same types to the *child* Route.
```

Comment:
Also under `grandparent`? That means in two different places we are referring to the same thing as "grandparent" which doesn't seem consistent with how family relations are described and may cause confusion.

Reply:
Agreed, this needs more work.